React/Fabric/RCTSurfacePresenter.mm

```diff
@@ -331,7 +331,10 @@ - (void)mountingManager:(RCTMountingManager *)mountingManager willMountComponent
 {
   RCTAssertMainQueue();
 
-  for (id<RCTSurfacePresenterObserver> observer in _observers) {
+  std::unique_lock<std::mutex> lock(_observerListMutex);
```

Comment:
We have to use `{ ... }` blocks to designate the scope covered by mutex. We must not call `.unlock` manually.
We also have to use better::shared_mutex and `std::shared_lock` here (and `std::unique_lock` where we modify the collection).

Reply:
Done, but seems may not give some benefits? 🤔  because reader seems always from main queue which mean only one reader. 😂